pallets/xcmp-handler/src/lib.rs

```diff
@@ -403,5 +414,125 @@ pub mod pallet {
 
 			Ok(().into())
 		}
+
+		/// Get total XCMP fees for all executions of an encoded call.
+		///
+		pub fn get_xcm_fee(
+			para_id: ParachainId,
+			currency_id: T::CurrencyId,
+			transact_encoded_call_weight: u64,
+		) -> Result<u128, DispatchError> {
+			let (fee, _weight) = Self::calculate_xcm_fee_and_weight(
+				para_id,
+				currency_id,
+				transact_encoded_call_weight,
+			)?;
+
+			Ok(fee)
+		}
+
+		/// Pay for XCMP fees.
+		/// Transfers fee from payer account to the local chain sovereign account.
+		///
+		pub fn pay_xcm_fee(source: T::AccountId, fee: u128) -> Result<(), DispatchError> {
+			let local_sovereign_account =
+				Sibling::from(T::SelfParaId::get()).into_account_truncating();
+
+			match T::Currency::transfer(
+				&source,
+				&local_sovereign_account,
+				<BalanceOf<T>>::saturated_from(fee),
+				frame_support::traits::ExistenceRequirement::KeepAlive,
+			) {
+				Ok(_number) => Self::deposit_event(Event::XcmFeesPaid {
+					source,
+					dest: local_sovereign_account,
+				}),
+				Err(e) => Self::deposit_event(Event::XcmFeesFailed {
+					source,
+					dest: local_sovereign_account,
+					error: e,
+				}),
+			};
+
+			Ok(().into())
+		}
+	}
+}
+
+pub trait XcmpTransactor<AccountId, CurrencyId> {
+	fn transact_xcm(
+		para_id: u32,
+		currency_id: CurrencyId,
+		caller: AccountId,
+		transact_encoded_call: sp_std::vec::Vec<u8>,
+		transact_encoded_call_weight: u64,
+	) -> Result<(), sp_runtime::DispatchError>;
+
+	fn get_xcm_fee(
+		para_id: u32,
+		currency_id: CurrencyId,
+		transact_encoded_call_weight: u64,
+	) -> Result<u128, sp_runtime::DispatchError>;
+
+	fn pay_xcm_fee(source: AccountId, fee: u128) -> Result<(), sp_runtime::DispatchError>;
+
+	#[cfg(feature = "runtime-benchmarks")]
+	fn setup_chain_currency_data(
+		para_id: u32,
+		currency_id: CurrencyId,
+	) -> Result<(), sp_runtime::DispatchError>;
+}
+
+impl<T: Config> XcmpTransactor<T::AccountId, T::CurrencyId> for Pallet<T> {
+	fn transact_xcm(
+		para_id: u32,
+		currency_id: T::CurrencyId,
+		caller: T::AccountId,
+		transact_encoded_call: sp_std::vec::Vec<u8>,
+		transact_encoded_call_weight: u64,
+	) -> Result<(), sp_runtime::DispatchError> {
+		Self::transact_xcm(
+			para_id.into(),
+			currency_id,
+			caller,
+			transact_encoded_call,
+			transact_encoded_call_weight,
+		)?;
+
+		Ok(()).into()
+	}
+
+	fn get_xcm_fee(
+		para_id: u32,
+		currency_id: T::CurrencyId,
+		transact_encoded_call_weight: u64,
+	) -> Result<u128, sp_runtime::DispatchError> {
+		let fee = Self::get_xcm_fee(para_id, currency_id, transact_encoded_call_weight)?;
```

Comment:
why not just use `calculate_xcm_fee_and_weight` instead of the new wrapper fn?

Reply:
makes sense.  I was taking in execution times and that's why I had a helper.

Reply:
done.